### DISCUSSION TITLE
Postal don't change destination IP automatically when get a SOFT FAIL

### DISCUSSION THREAD

**Discussion by lvaro22** [postalserver/postal#2630] [Help with using Postal]
I have an SMTP with 2 IPs for redundance on my domain.

If a soft fail is detected on the first IP, POSTAL always tries the same IP again and doesn't switch to the second one.

DNS configuration is correct, as confirmed by successful connection through telnet because when detects a fail on the first IP, try the second one inmediately.
![imagen](https://github.com/postalserver/postal/assets/60357708/95dcbd61-2909-47bc-9a7e-9cb6972c3cda)

POSTAL is not behaving the same way and is getting stuck on the same IP, causing messages to be dropped into the queue with soft fail detected.

I appreciate your attention.

**Comment by willpower232** [comment]
In case Postal is asking the wrong thing for DNS, you could try changing this setting (and restarting Postal)

https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/config/postal.defaults.yml#L17

Otherwise unfortunately it seems to be an issue. Are you presenting multiple IPs on the same A record for `mail.test.es`? I don't know if there needs to be multiple MX records, one for each IP address.

**Reply by lvaro22** [reply]
Thanks for responding.

I tried different configurations, two MX with the same priority for balance and only one MX with two A records too, but none of them worked. The DNS configuration is fine because It works well on other platforms.

It seems that POSTAL retrieves the IP address from the fields of a SOFT FAIL, or a similar mechanism (possibly involving cache memory?), as it consistently tries to use the same IP and does not verify DNS records when retrying to send messages.

Are there any plans to fix this issue in upcoming versions?

thank you once again.

**Reply by lvaro22** [reply]
I also attempted to modify the setting `use_local_ns_for_domains: false` too, but it didn't work.

**Reply by willpower232** [reply]
I'm pretty sure I've seen Postal work its way through MX records before but I could be wrong, have you tried with different priorities? Are you giving each IP its own domain name when using multiple MX records?

Unless someone provides the code change, its unlikely anyone is working on this as they tend not to make changes if Postal is working for them and it is for the majority I am afraid.
